commit 59e049b5c4adefe488808bb11009ea585b0c7a6a (refs/changes/59/2159/2)
Author: David Pacheco <dap@joyent.com>
Date:   2017-06-27T16:55:36-07:00 (2 years, 3 months ago)
    
    MANTA-3333 postgres long-running query probe should exclude autovacuum
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/alarm_metadata/probe_templates/postgres.yaml b/alarm_metadata/probe_templates/postgres.yaml
index 04659cd..b661696 100644
--- a/alarm_metadata/probe_templates/postgres.yaml
+++ b/alarm_metadata/probe_templates/postgres.yaml
@@ -235,7 +235,7 @@
         -
             type: cmd
             config:
-                cmd: "/opt/local/bin/psql -U postgres -c \"SELECT *, NOW() - query_start AS elapsed FROM pg_stat_activity WHERE state != 'idle' AND NOW() - query_start > interval '120 seconds'\""
+                cmd: "/opt/local/bin/psql -U postgres -c \"SELECT *, NOW() - query_start AS elapsed FROM pg_stat_activity WHERE state != 'idle' AND query NOT LIKE 'autovacuum: %' AND NOW() - query_start > interval '120 seconds'\""
                 interval: 60
                 timeout: 30
                 stdoutMatch:
